File: 2014m./uzd2./sprendimai/uzd2.py
class Path:
    path = []
    length = 0
    result = ""

def main():
    data = open("U2.txt", "r")
    output = open("U2rez.txt", "w")

    temp = next(data).split(" ")
    temp = [int(o) for o in temp]
    startPosX = temp[0]
    startPosY = temp[1]

    temp = next(data).split(" ")
    temp = [int(o) for o in temp]
    endPosX = temp[0]
    endPosY = temp[1]

    pathCount = int(next(data))
    paths = [None]*pathCount

    for i in range(pathCount):
        posX = startPosX
        posY = startPosY

        pathData = next(data).split(" ")
        pathData = [int(o) for o in pathData]
        pathTaken = []
        pathLength = 0
        pathResult = "sekos pabaiga       "

        for j in range(1, pathData[0] + 1):
            if(pathData[j] == 1):
                posY += 1
            elif(pathData[j] == 2):
                posX += 1
            elif(pathData[j] == 3):
                posY -= 1
            else:
                posX -= 1
            pathLength += 1
            pathTaken.append(str(pathData[j]))
            if(posX == endPosX and posY == endPosY):
                pathResult = "pasiektas tikslas   "
                break
        
        path = Path()
        path.path = pathTaken
        path.length = pathLength
        path.result = pathResult
        
        paths[i] = path

    for path in paths:
        output.write(f"{path.result} {' '.join(path.path)} {str(path.length)}\n")

    data.close()
    output.close()

if __name__ == "__main__":
    main()

File: 2014m./uzd2./sprendimai/uzd2.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <vector>
#include <string>

using namespace std;

struct Path {
    vector<int> path;
    int length;
    string result;
};

int main() {
    ifstream data("U2.txt");
    ofstream output("U2rez.txt");

    int startPosX, startPosY, endPosX, endPosY;
    data >> startPosX >> startPosY;
    data >> endPosX >> endPosY;

    int pathCount;
    data >> pathCount;
    vector<Path> paths(pathCount);

    data.ignore();

    for (int i = 0; i < pathCount; ++i) {
        int posX = startPosX;
        int posY = startPosY;

        string line;

        getline(data, line);
        istringstream datasream(line);

        int pathDataSize;
        datasream >> pathDataSize;
        vector<int> pathData(pathDataSize);

        paths[i].result = "sekos pabaiga       ";

        for (int j = 0; j < pathDataSize; ++j) {
            datasream >> pathData[j];
            if (pathData[j] == 1) {
                posY += 1;
            } else if (pathData[j] == 2) {
                posX += 1;
            } else if (pathData[j] == 3) {
                posY -= 1;
            } else {
                posX -= 1;
            }

            paths[i].path.push_back(pathData[j]);
            paths[i].length++;

            if (posX == endPosX && posY == endPosY) {
                paths[i].result = "pasiektas tikslas   ";
                break;
            }
        }
    }

    for (const Path& path : paths) {
        output << path.result << " ";
        for (int point : path.path) {
            output << point << " ";
        }
        output << path.length << "\n";
    }

    data.close();
    output.close();
}

